bbq/compiler/desugar.go

```diff
@@ -533,12 +533,115 @@ func (d *Desugar) desugarCondition(condition ast.Condition) ast.Statement {
 		return ifStmt
 	case *ast.EmitCondition:
 		emitStmt := (*ast.EmitStatement)(condition)
-		return emitStmt
+
+		if !isInherited {
+			return emitStmt
+		}
+
+		// If the condition is inherited, then re-write
+		// the emit statement to be type-qualified.
+		// i.e: `emit Event()` will be re-written as `emit Contract.Event()`.
+		// Otherwise, the compiler can't find the symbol.
```

Comment:
I'm not supper happy having to re-write it this way. If there are better ways to do it, we should definitely explore those. I am out of ideas though.